modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -1370,7 +1370,10 @@
   "Parse a serialized native query. Like a normal JSON parse, but handles BSON/MongoDB extended JSON forms."
   [^String s]
   (try
-    (mapv (fn [^org.bson.BsonValue v] (-> v .asDocument com.mongodb.BasicDBObject.))
+    ;; TODO: Fixme! In following expression we were previously creating BasicDBObject's. As part of Monger removal
+    ;;       in favor of plain mongo-java-driver we now create Documents. I believe following conversion was and is
+    ;;       responsible for #38181.
```

Comment:
Can you make this more specific? Reading this, I'm not quite sure what is to be fixed (and how).

Reply:
Done.